### ISSUE TITLE
TypeError: 'str' does not support the buffer interface at pdfinterp.execute

### ISSUE THREAD

**Issue by 0xabu** [pdfminer/pdfminer.six#23]
I'm trying to migrate some pdfminer code to python3 (which was working with the upstream pdfminer on python2.7) using this version of pdfminer. It fails on:

```
  File "mycode.py", line 123, in main
    interpreter.process_page(page)
  File "pdfminer/pdfinterp.py", line 834, in process_page
    self.render_contents(page.resources, page.contents, ctm=ctm)
  File "pdfminer/pdfinterp.py", line 846, in render_contents
    self.execute(list_value(streams))
  File "pdfminer/pdfinterp.py", line 870, in execute
    func(*args)
  File "pdfminer/pdfinterp.py", line 811, in do_Do
    interpreter.render_contents(resources, [xobj], ctm=mult_matrix(matrix, self.ctm))
  File "pdfminer/pdfinterp.py", line 846, in render_contents
    self.execute(list_value(streams))
  File "pdfminer/pdfinterp.py", line 862, in execute
    method = 'do_%s' % name.replace('*', '_a').replace('"', '_w').replace("'", '_q')
TypeError: 'str' does not support the buffer interface
```

This seems to be a string conversion issue due to the following new code in psparser.py:

```
def keyword_name(x):
    if not isinstance(x, PSKeyword):
        # (snip)
    else:
        name=x.name
        if six.PY3:
            try:
                name = str(name,'utf-8')
            except:
                pass
    return name
```

Sticking a 'raise' in there (rather than pass) shows that the utf-8 decoding is failing ("invalid start byte"), and indeed the name looks like binary junk. There are a lot of these bad keyword names in this particular PDF, and they're all on the same page, so it may well be a malformed PDF or a parser bug elsewhere. (Sorry, I can't share the PDF.) Nevertheless, pdfminer should probably be able to handle this more robustly, because these bad names would have been ignored by execute() if STRICT was off, which it is by default in the original pdfminer.

So, I have two sub-buglets here (sorry for lumping them together):
- we shouldn't assume that every keyword name can be decoded as utf-8; maybe you could pass errors='ignore' to the str() conversion, or at least return something that's a valid string type, rather than returning the underlying bytes
- settings.STRICT should probably remain off by default if you don't want to break existing code

**Comment by setivolkylany** [comment]
For me it is not working too.

Code:
```
import sys
import platform

print('Python detail: ', sys.version_info)
print('System info:', platform.platform())

from pdfminer.pdfdocument import PDFDocument
from pdfminer.pdfparser import PDFParser
from pdfminer.pdfinterp import PDFResourceManager, PDFPageInterpreter
from pdfminer.converter import TextConverter
from pdfminer.layout import LAParams
from pdfminer.pdfpage import PDFPage

with open('/home/wlysenko/Downloads/Report about polls 2016-08-18 13-14-11.pdf', 'rb') as f:
    parserpdf = PDFParser(f)
    doc = PDFDocument(parserpdf)
    h = io.BytesIO()
    manager = PDFResourceManager()
    converter = TextConverter(manager, h, codec='utf8', laparams=LAParams())
    interpreter = PDFPageInterpreter(manager, converter)

    for page in PDFPage.get_pages(f, {1}):
        interpreter.process_page(page)
```

Console:

```
Python detail:  sys.version_info(major=3, minor=4, micro=3, releaselevel='final', serial=0)
System info: Linux-3.13.0-37-generic-x86_64-with-LinuxMint-17.1-rebecca
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-41-83f8d62b1f63> in <module>()
     22 
     23     for page in PDFPage.get_pages(f, {1}):
---> 24         interpreter.process_page(page)

/home/wlysenko/.virtualenvs/programmerHelper/lib/python3.4/site-packages/pdfminer/pdfinterp.py in process_page(self, page)

/home/wlysenko/.virtualenvs/programmerHelper/lib/python3.4/site-packages/pdfminer/pdfinterp.py in render_contents(self, resources, streams, ctm)

/home/wlysenko/.virtualenvs/programmerHelper/lib/python3.4/site-packages/pdfminer/pdfinterp.py in execute(self, streams)

TypeError: 'str' does not support the buffer interface
```

**Comment by goulu** [comment]
Hi @0xabu . I see you solved this problem and others on your fork. Could you please merge your code here with a pull request ? BTW I invited you to the group who now "owns" pdfminer.six ...
Thanks !

**Comment by 0xabu** [comment]
Doing exactly that has been on my TODO list for a few months now. Sorry :) I will try to get you a pull request next week.

**Comment by vstoykov** [comment]
This looks like is fixed in https://github.com/pdfminer/pdfminer.six/commit/9439a3a31a347836aad1c1226168156125d9505f right? then I suppose that the issue could be closed.

**Comment by 0xabu** [comment]
Yep, fixed in #47

**Comment by alanzchen** [comment]
I met the same problem with pdfminer.six-20160614 on PyPI.

Obviously the version on PyPI has not been updated, right?
